### PR TITLE
Rename `IP_Unix`, `IP_Address` and `TCP_Server` to remove underscores

### PR DESCRIPTION
**Pull request by Calinou**
See #16863.

### CODE REVIEW

File: drivers/unix/ip_unix.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ static IP_Address _sockaddr2ip(struct sockaddr *p_addr) {
 	return ip;
 };
 
-IP_Address IP_Unix::_resolve_hostname(const String &p_hostname, Type p_type) {
+IP_Address IPUnix::_resolve_hostname(const String &p_hostname, Type p_type) {
```
Comment (akien-mga) [inline drivers/unix/ip_unix.cpp:93]: `IP_Address` should likely be `IPAddress`.

**Comment by pycbouh** [comment]
Since renames are being merged these days, maybe worth rebasing this one? Was reminded about this horrible naming scheme and had to double check if we'd already fixed it 🙂

**Comment by Calinou** [comment]
> Since renames are being merged these days, maybe worth rebasing this one? Was reminded about this horrible naming scheme and had to double check if we'd already fixed it :slightly_smiling_face:

Rebased.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**